Blog post for sprint 128 - request redesign (#382)
openSUSE/obs-landing: _posts/development/2022-10-24-request-workflow-redesign.md

---
layout: post
title: Request Page Redesign - Build results and superseded conversations
category: development
---

> We don't stop improving the requests workflow and we are actively working on the feedback. This time we worked on including conversations from superseded requests, and presenting build results.

Reviewer: ```suggestion
We don't stop improving the requests workflow and we are actively working on the feedback. This time we worked on including conversations from superseded requests and presenting build results.
```

The request redesign is part of [the beta program](/2018/10/04/the-beta-program/).

{% include partials/_series-of-posts-about-request-redesign.md %}

## Build results

> Build results are one of the most important features for the OBS users. They are specifically important for requests because they help you to verify if your changes are working or not. Now we present build results in a separate tab on request page.

Reviewer 2: ```suggestion
Build results are specifically important for requests because they help you to verify if the submitted changes are working or not. Now we present build results in a separate tab on request page.
```

<figure>

>   <img src="/images/posts/build-results.png" alt="A screenshot of the build results" />

Reviewer 2: Make sure you update the screenshot to the latest state.

Author: It's updated

Reviewer 2: This is still not it. Guess you are looking at production, I don't think it's deployed yet.

  <figcaption>Buils results</figcaption>
</figure>

## Superseded conversations

> In the old design the conversation of a superseded request was presented in a tab. Because the conversation on the current request is very often "just" the continuation of people talking about the previous request. We brought this back and integrated the conversation right into the stream (collapsed by default).

Reviewer: ```suggestion
In the old design, the conversation of a superseded request was presented in a tab. Because the conversation on the current request is very often "just" the continuation of people talking about the previous request. We brought this back and integrated the conversation right into the stream (collapsed by default).
```

> If you create a superseding request, conversation and history of the superdeded request was not visible. The only possible way to check the history was to visit the superseded request page. Now we have improved it and the history from superseded requests is now directly available in the conversation tab of superseding requests.

Reviewer 2: ```suggestion
In the old design the conversation of a superseded request was presented in a tab. Because the conversation on the current request is very often "just" the continuation of people talking about the previous request. We brought this back and integrated the conversation right into the stream (collapsed by default).
```

> If you create a request superseding request, conversation and history of the superdeded request was not visible. The only possible way to check the history was to visit the superseded request page. Now we have improved it and the history from superseded requests is now available in the superseding requests.

Author: Right, I was referring to the new view.

---
layout: post
title: Request Page Redesign - Build results, superseded conversations and diff URLs
category: development
---

We don't stop improving the requests workflow and we are actively working on the feedback. This time we worked on including conversations from superseded requests, sharing diff URL's and presenting build results.

The request redesign is part of [the beta program](/2018/10/04/the-beta-program/).

{% include partials/_series-of-posts-about-request-redesign.md %}

## Superseded conversations

> There was no way to know the history and conversations happened in a superseded requests. The only possible way to check the history was to visit the superseded request page. Now we have improved it and the history from superseded requests is now available in the superseding requests.

Reviewer 2: You are describing the change from first iteration to this now. I think what you should describe is the change from what was there before on the old request page to how it's done now.

Author: Its updated

> There was no way to know the history and conversations happened in a superseded requests. The only possible way to check the history was to visit the superseded request page. Now we have improved it and the history from superseded requests is now available in the requests that supersedes other requests.

Reviewer 3: ```suggestion
There was no way to know the history and conversations happened in a superseded requests. The only possible way to check the history was to visit the superseded request page. Now we have improved it and the history from superseded requests is now available in the superseding requests.
```

<figure>
  <img src="/images/posts/superseded-conversations.png" alt="A screenshot of the conversation" />
  <figcaption>Conversation and history from superseded request</figcaption>
</figure>

## Diff URLs

> A very common usecase when working with requests is to pointout a change in diffs. For example if someone has removed or added something new in the code. Now it's very easy to point to that change by a unique URL. All you have to do is click on line number of the code.

Reviewer 2: This is actually a bug fixed throughout the app, not very specific to this opportunity. And for sure not more important than build results. I would mention it at the end, if at all...

Author: Okay, I've removed it

<figure>
  <img src="/images/posts/diff-url.png" alt="A screenshot of the highlighted diff" />
  <figcaption>Diff URL</figcaption>
</figure>

## Build results

> Build results are one of the most important features for the OBS users. Now the request page presents build results in a separate tab.

Reviewer 2: Why is it important? And why is it important in the Request context? You should spell that out.

> Build results are one of the most important features for the OBS users. Now the request page presents build results in it's separate tab.

Reviewer 3: ```suggestion
Build results are one of the most important features for the OBS users. Now the request page presents build results in a separate tab.
```
